lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -421,7 +468,32 @@ computeTxSize proxy action cs =
           where
             chaff = L8.pack (show ix) <> BL.fromStrict txid
 
-    wits = SL.WitnessSet (Set.union addrWits certWits) mempty mempty
+    byronWits = Set.map dummyWitnessUniq $ Set.fromList (fst <$> CS.inputs cs)
+      where
+        dummyWitness :: BL.ByteString -> SL.BootstrapWitness TPraosStandardCrypto
+        dummyWitness chaff =
+            SL.makeBootstrapWitness credential signingKey addrAttr
+          where
+            credential = Hash.UnsafeHash $ L8.toStrict chaff
+            addrAttr = Byron.mkAttributes $ Byron.AddrAttributes
+                       (Just $ Byron.HDAddressPayload (bloatChaff keyLen))
+                       (toByronNetworkMagic mainnetMagic)
```

Comment:
mainnet is hard-coded here which is sort of wrong because mainnet addresses don't embed any discrimination attribute whereas testnet address do. So there's a significant size variation between the two.

Reply:
suggestion: rely on the `proxy` above to know whether mainnet or testnet is needed. See also: `fromNetworkDiscriminant`

Reply:
can I retrieve protocolMagic easily from proxy? :thinking: 
I have above (I agree it was quick hack) : 
```
763 toByronNetworkMagic :: W.ProtocolMagic -> Byron.NetworkMagic                                                                                                                                                   
764 toByronNetworkMagic pm@(W.ProtocolMagic magic) =                                                                                                                                                               
765     if pm == W.mainnetMagic then                                                                                                                                                                               
766         Byron.NetworkMainOrStage                                                                                                                                                                               
767     else                                                                                                                                                                                                       
768         Byron.NetworkTestnet (fromIntegral magic)
```

Reply:
In theory they should match indeed.